hexrd/instrument/hedm_instrument.py

```diff
@@ -2052,23 +2042,18 @@ def pull_spots(self, plane_data, grain_params,
                             patch_data = patch_data_raw
                             pass  # end contains_signal
                         # write output
-                        if filename is not None:
-                            if output_format.lower() == 'text':
-                                writer.dump_patch(
-                                    peak_id, hkl_id, hkl, sum_int, max_int,
-                                    ang_centers[i_pt], meas_angs,
-                                    xy_centers[i_pt], meas_xy)
-                            elif output_format.lower() == 'hdf5':
-                                xyc_arr = xy_eval.reshape(
-                                    prows, pcols, 2
-                                ).transpose(2, 0, 1)
-                                writer.dump_patch(
-                                    detector_id, iRefl, peak_id, hkl_id, hkl,
-                                    tth_edges, eta_edges, np.radians(ome_eval),
-                                    xyc_arr, ijs, frame_indices, patch_data,
-                                    ang_centers[i_pt], xy_centers[i_pt],
-                                    meas_angs, meas_xy)
-                            pass  # end conditional on write output
+                        xyc_arr = xy_eval.reshape(
+                            prows, pcols, 2
+                        ).transpose(2, 0, 1)
+                        spot = Spot(
+                            detector_id, iRefl, peak_id, hkl_id, hkl,
+                            tth_edges, eta_edges, np.radians(ome_eval),
+                            xyc_arr, ijs, frame_indices, patch_data,
+                            ang_centers[i_pt], xy_centers[i_pt],
+                            meas_angs, meas_xy, sum_int, max_int
```

Comment:
When there are many arguments, I think it's usually preferable to use kwargs, since it is less error-prone (if you forget a kwarg, the error is clearer than if you forget an arg, and you don't have to worry about accidentally swapping two arguments if you use kwargs, which can lead to subtle bugs if two args are similar).

But since this is already using args, we can just leave it!